File: martianrun/core/src/com/gamestudio24/martianrun/actors/menu/AboutButton.java
package com.gamestudio24.martianrun.actors.menu;

import com.badlogic.gdx.math.Rectangle;
import com.gamestudio24.martianrun.enums.GameState;
import com.gamestudio24.martianrun.utils.Constants;
import com.gamestudio24.martianrun.utils.GameManager;

public class AboutButton extends GameButton {

    public interface AboutButtonListener {
        void onAbout();
    }

    private final AboutButtonListener listener;

    public AboutButton(Rectangle bounds, AboutButtonListener listener) {
        super(bounds);
        this.listener = listener;
    }

    @Override
    protected String getRegionName() {
        return GameManager.getInstance().getGameState() == GameState.ABOUT ? Constants.CLOSE_REGION_NAME :
                Constants.ABOUT_REGION_NAME;
    }

    @Override
    public void touched() {
        listener.onAbout();
    }

}

File: martianrun/core/src/com/gamestudio24/martianrun/actors/menu/AboutButton.kt
package com.gamestudio24.martianrun.actors.menu

import com.badlogic.gdx.math.Rectangle
import com.gamestudio24.martianrun.enums.GameState
import com.gamestudio24.martianrun.utils.Constants
import com.gamestudio24.martianrun.utils.GameManager.Companion.instance

class AboutButton(bounds: Rectangle, private val listener: AboutButtonListener) : GameButton(bounds) {
    interface AboutButtonListener {
        fun onAbout()
    }

    override fun getRegionName(): String {
        return if (instance.gameState == GameState.ABOUT) Constants.CLOSE_REGION_NAME else Constants.ABOUT_REGION_NAME
    }

    override fun touched() {
        listener.onAbout()
    }
}
